src/merkle/tiered_smt/tests.rs

```diff
@@ -32,6 +30,17 @@ fn tsmt_insert_one() {
     // make sure the paths we get from the store and the tree match
     let expected_path = store.get_path(tree_root, index).unwrap();
     assert_eq!(smt.get_path(index).unwrap(), expected_path.path);
+
+    // make sure inner nodes match
+    let expected_nodes = get_non_empty_nodes(&store);
+    let actual_nodes = smt.inner_nodes().collect::<Vec<_>>();
```

Comment:
Shouldn't we consider using `MerkleStore` as backend of the implementation so it will be consistent with the other trees? It will also make it easier in the future to move from in-memory only to more robust, large scale storage solutions

Reply:
The current `MerkleStore` will probably remain an in-memory structure. I'm thinking we'll create specialized structures for cases when we need to handle large amounts of data. For example, we could create specialized structures for `AccountDB` and `NullifierDB` which take advantage of special properties of these constructs.